scripts/append_metadata.cmake

```diff
@@ -0,0 +1,68 @@
+cmake_minimum_required(VERSION 3.5)
+
+# Usage: cmake -DEXTENSION=path/to/extension.duckdb_extension -DPLATFORM_FILE=README.md -DDUCKDB_VERSION=tag1 -DEXTENSION_VERSION=tag2 -P scripts/append_metadata.cmake
+# Currently hardcoded to host up to 8 fields
+# Example: ./scripts/append_metadata.sh file.duckdb_extension git_hash_duckdb_file git_hash_extension_file platfrom_file
```

Comment:
nit: this still say append_metadata.sh

Also with CMake's very weird partial support for null bytes (requiring this horrible `null.txt` hack), i would like to have some double checks here that our header is written correctly. Maybe we can add a check afterwards that confirms the header is correct. 

I will add this in a follow-up PR to not block this one

Reply:
Will fix those in a follow up.